controllers/authControllers.js

```diff
@@ -0,0 +1,120 @@
+const { promisify } = require('util')
+const jwt = require('jsonwebtoken')
+const User = require('../models/userModel')
+
+const signToken = id => {
+    return jwt.sign(
+        { id },
+        process.env.JWT_SECRET,
+        {
+            expiresIn: process.env.JWT_EXPIRES_IN
+        }
+    )
+}
+
+exports.signup = async (req, res) => {
+    try {
+        const newUser = await User.create({
+            name: req.body.name,
+            email: req.body.email,
+            password: req.body.password,
+            passwordConfirm: req.body.passwordConfirm
+        })
+
+        const token = signToken(newUser._id)
+
+        res.status(201).json({
+            status: 'success',
+            token,
+            data: { newUser }
+        })
+    } catch (error) {
+        res.status(404).json({
+            status: 'fail',
+            message: error
+        })
+    }
+}
+
+exports.login = async (req, res, next) => {
+    const { email, password } = req.body
+
+    //Check if email and password exist
+    if (!email || !password) {
+        return next(
+            res.status(400).json({
+                status: 'failed',
+                message: 'Please provide a valid email and password'
+            })
+        )
+    }
+
+    //Check is user exist && Password is correct
+    const user = await User.findOne({ email: email }).select('+password')
+
+    if (!user || !(await user.correctPassword(password, user.password))) {
+        return next(
+            res.status(401).json({
+                status: 'failed',
+                message: 'Incorrect email and password'
```

Comment:
Is it 'Incorrect email and password' or 'Incorrect email or password' Look at your if statement on line 55.